docs/en/lua-developer-faq.md

```diff
@@ -311,35 +378,16 @@ Note that if you use `require("XXX")` to load your code then this will automatic
+Functions have fixed overheads, so in general the more that your group your application code into larger function, then  less RAM used will be used overall.  The main caveat here is that if you are starting to do "copy and paste" coding across functions then you are wasting resources.  So of course you should still use functions to structure your code and encapsulate common repeated processing, but just bear in mind that each function definition has a relatively high overhead for its header record and stack frame.  *So try to avoid overusing functions. If there are less than a dozen or so lines in the function then you should consider putting this code inline if it makes sense to do so.*
 
 ### What other resources are available?
 
 * Install lua and luac on your development PC.  This is freely available for Windows, Mac and Linux distributions, but we strongly suggest that you use Lua 5.1 to maintain source compatibility with ESP8266 code.  This will allow you not only to unit test some modules on your PC in a rich development environment, but you can also use `luac` to generate a bytecode listing of your code and to validate new code syntactically before downloading to the ESP8266.  This will also allow you to develop server-side applications and embedded applications in a common language. 
 
 ## Firmware and Lua app development
 
-### How to save memory?
-* The NodeMCU development team recommends that you consider using a tailored firmware build, which only includes the modules that you plan to use before developing any Lua application. Once you have the ability to make and flash custom builds, the you also have the option of moving time sensitive or logic intensive code into your own custom module. Doing this can save a large amount of RAM as C code can be run directly from Flash memory. If you want an easy-to-use intermediate option then why note try the [cloud based NodeMCU custom build service](https://nodemcu-build.com)?. 
+### How to reduce the size of the firmware?
+
+* We recommend that you consider using a tailored firmware build; one which only includes the modules that you plan to use in developing any Lua application. Once you have the ability to make and flash custom builds, the you also have the option of moving time sensitive or logic intensive code into your own custom module. Doing this can save a large amount of RAM as C code can be run directly from Flash memory. If you want an easy-to-use intermediate option then why note try the [cloud based NodeMCU custom build service](https://nodemcu-build.com). 
```

Comment:
Yup, since the FAQ is written in a generally personal voice, the shift to first person works; however,   the grammar still isn't quite right here, so I'll update accordingly and reference the build section.  

One of the thing that has changes since V1 is that it's primary host was esp8266.com and decent documentation didn't exist.  I have tried to integrate intra-documentation references and avoid too much duplication when practical.